application-task-manager-ui/src/main/resources/TaskManager/Code/TaskReportMacro.xml

```diff
@@ -0,0 +1,1013 @@
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity output="false"}}
+#macro(addToExtraTags $paramValues $paramName)
+  #foreach ($param in $paramValues.split('(%20)*%2C(%20)*'))
+    #set ($extraParams = "$!{extraParams}&amp;${paramName}=${param}")
+  #end
+#end
+#macro(addReferenceToExtraTags $list $paramName)
+  #foreach ($item in $list)
+    #set ($extraParams = "$!{extraParams}&amp;${paramName}=${escapetool.url($item.reference)}")
+  #end
+#end
+{{/velocity}}
+{{velocity}}
+#set ($spaces = $escapetool.url($wikimacro.parameters.spaces))
+#set ($tags = $escapetool.url($wikimacro.parameters.labels))
+#set ($assignees = $wikimacro.parameters.assignees)
+#set ($creator = $wikimacro.parameters.creators)
+#set ($createdAfter = $escapetool.url($xcontext.macro.params.createddateFrom))
+#set ($status = $escapetool.url($wikimacro.parameters.status))
+#set ($pageSize = $escapetool.url($wikimacro.parameters.pageSize))
+#set ($sortBy = $escapetool.url($wikimacro.parameters.sortBy))
+#set ($displayColumns = $escapetool.url($wikimacro.parameters.columns))
+#set ($rowCount = $wikimacro.parameters.rowCount)
+#set ($columns = $displayColumns.split('\s*%2C\s*'))
+#set ($extraParams = "&amp;creator=$!{escapetool.url($creator.reference)}&amp;createDate=$!{createdAfter}&amp;status=$!{status}")
+#set ($rowCount = $numbertool.toNumber($rowCount).intValue())
+#addToExtraTags($tags 'tag')
+#addToExtraTags($spaces 'space')
+#addReferenceToExtraTags($assignees 'taskAssignee')
+#addReferenceToExtraTags($creator 'taskCreator')
+#set ($columnsProperties = {
+  'description': {'type': 'text', 'size': 10, 'html': true},
+  'assignee': {'type': 'text', 'size': 10, 'html': true},
+  'createDate': {'type': 'date', 'size': 10},
+  'deadline': {'type': 'text', 'size': 10},
+  'status': {'type': 'list', 'size': 10},
+  'doc.title' : {'type': 'text', 'size': 10, 'link': 'view'}
+})
+#set ($options = {
+  'resultPage': 'TaskManager.Code.TaskReportResultPage',
+  'className': 'TaskManager.Code.TaskClass',
+  'translationPrefix': 'taskmanager.livetable.',
+  'tagCloud': true,
+  'selectedColumn': $!sortBy,
+  'rowCount': $rowCount,
+  'extraParams': "${extraParams}"
+})
+#livetable('tasks' $columns $columnsProperties $options)
```

Comment:
You have deal with multiple calls of the task report macro on the same page. Maybe generate an id with the "tasks" prefix, or better use a macro parameter (id), with a fallback on the generated id.